capstone/src/main/java/com/google/sps/servlets/BlobstoreUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public final class BlobstoreUtil {
+  /* Return BlobKey for image uploaded through form. */
+  public BlobKey getBlobKey(
+      HttpServletRequest request, String formInputElementName, BlobstoreService blobstoreService) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+    return blobKey;
+  }
+
+  public void doGetHelper(
```

Comment:
Does this work? I'm not exactly sure how the parameter objects (HttpServletResponse, BlobstoreService, etc.) work, but the operations done need to persist right? That probably indicates this method shouldn't be static and in general, Util classes should only contain static public methods. So it might be better to keep this within the individual servlets (even if it's a bit redundant).

Reply:
Ah, I see this makes sense. Just moved the doGetHelper method back to each file, thank you for bringing this up!